src/main/java/com/spotify/docker/client/npipe/NamedPipeSocket.java

```diff
@@ -0,0 +1,308 @@
+/*-
+ * -\-\-
+ * docker-client
+ * --
+ * Copyright (C) 2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.docker.client.npipe;
+
+import java.io.FilterInputStream;
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.RandomAccessFile;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.SocketException;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.channels.SocketChannel;
+
+public class NamedPipeSocket extends Socket {
+
+  private final Object connectLock = new Object();
+  private volatile boolean inputShutdown;
+  private volatile boolean outputShutdown;
+
+  private String socketPath;
+  private volatile SocketAddress socketAddress;
+  private RandomAccessFile namedPipe;
+
+  private FileChannel channel;
+
+  NamedPipeSocket() throws IOException {
+  }
+
+  @Override
+  public void connect(SocketAddress endpoint) throws IOException {
+    connect(endpoint, 0);
+  }
+
+  @Override
+  public void connect(SocketAddress endpoint, int timeout) throws IOException {
+    if (timeout < 0) {
+      throw new IllegalArgumentException("Timeout may not be negative: " + timeout);
+    }
+
+    if (!(endpoint instanceof NpipeSocketAddress)) {
+      throw new IllegalArgumentException("Unsupported address type: " 
+        + endpoint.getClass().getName());
+    }
+
+    this.socketAddress = endpoint;
+    this.socketPath = ((NpipeSocketAddress) endpoint).path();
+
+    synchronized (connectLock) {
+      this.namedPipe = new RandomAccessFile(socketPath, "rw");
+      this.channel = this.namedPipe.getChannel();
+    }
+  }
+
+  @Override
+  public void bind(SocketAddress bindpoint) throws IOException {
+    throw new SocketException("Bind is not supported");
+  }
+
+  @Override
+  public InetAddress getInetAddress() {
+    return null;
```

Comment:
Would throwing `UnsupportedOperationException` in these methods be clearer instead?